django/db/backends/mysql/schema.py

```diff
@@ -135,16 +155,71 @@ def _field_should_be_indexed(self, model, field):
             return False
         return not self._is_limited_data_type(field)
 
-    def _delete_composed_index(self, model, fields, *args):
+    def _create_missing_fk_index(
+        self,
+        model,
+        *,
+        fields=None,
+        expressions=None,
+        name=None,
+    ):
         """
         MySQL can remove an implicit FK index on a field when that field is
         covered by another index like a unique_together. "covered" here means
         that the more complex index starts like the simpler one.
-        https://bugs.mysql.com/bug.php?id=37910 / Django ticket #24757
-        We check here before removing the [unique|index]_together if we have to
+        https://bugs.mysql.com/bug.php?id=37910 / Django tickets #24757,31335
+        We check here before removing the composed index if we have to
         recreate a FK index.
         """
-        first_field = model._meta.get_field(fields[0])
+
+        def get_first_index_field_name(index_fields, index_expressions):
+            field = None
+            if index_fields:
+                if (
+                    index_fields[0].startswith("-")
+                    and not self.connection.features.supports_index_column_ordering
+                ):
+                    field = index_fields[0][1:]
+                elif not index_fields[0].startswith("-"):
+                    field = index_fields[0]
```

Comment:
If I understand correctly, when the first field is indexed in descending order (like `-fieldname`) and the DB supports it, we will return `None` for this function.
1/ MySQL doesn't drop the implicit FK index when another index has as first field the FK in descending order, is that correct?
2/ The naming of the function is not _entirely_ accurate with respect to the behaviour, since we'll return None even though one would expect to get the first field name.